Projects/BookJSON2/main.go

```diff
@@ -30,26 +31,39 @@ var books []Book
 func (b *Books) book(w http.ResponseWriter, r *http.Request) {
 	// if r.Method == POST
 	switch r.Method {
-	case "POST": //Retriving a book with Id number
+	case "GET": //Retriving a book with Id number
 		{
+
 			w.Header().Set("content-type", "application/json")
-			jsonreqestfromweb := &Book{}
-			err := json.NewDecoder(r.Body).Decode(&jsonreqestfromweb)
+			val := r.FormValue("id")
+			id, err := strconv.Atoi(val)
 			if err != nil {
-				fmt.Println(err)
-				return
+				panic(err)
+			}
+			if id >= len(b.Books) {
+				w.WriteHeader(http.StatusNotFound)
+
+			} else {
+				json.NewEncoder(w).Encode(b.Books[id])
 			}
 
-			json.NewEncoder(w).Encode(b.Books[jsonreqestfromweb.Id])
-			break
 		}
 	case "PUT": // adding new book
```

Comment:
It is expected at REST that

- POST is for adding new items
- PUT is for modifying existing ones
- PATCH is optional (I would not implement that here) and is for modifying existing ones providing only instructions for the modified fields.

I would expect a POST for adding new books.